Gems/ROS2/Code/Source/Camera/PostProcessing/ROS2ImageEncodingConvertComponent.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2ImageEncodingConvertComponent.h"
+#include <AzCore/RTTI/RTTIMacros.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <sensor_msgs/msg/detail/image__struct.hpp>
+
+namespace AZStd
+{
+    template<>
+    struct hash<ROS2::EncodingConverion>
+    {
+        size_t operator()(const ROS2::EncodingConverion& data) const
+        {
+            return (static_cast<AZ::u16>(data.encodingIn) << 8) | static_cast<AZ::u16>(data.encodingOut);
+        }
+    };
+} // namespace AZStd
+
+namespace ROS2
+{
+    namespace
+    {
+        void Rgba8ToRgb8(sensor_msgs::msg::Image& image)
+        {
+            AZ_Assert(image.encoding == "rgba8", "Image encoding is not rgba8");
+            AZ_Assert(image.step == image.width * 4, "Image step is not width * 4");
+            AZ_Assert(image.data.size() == image.step * image.height, "Image data size is not step * height");
+
+            // Perform conversion in place
+            for (size_t pixelId = 0; pixelId < image.width * image.height; ++pixelId)
+            {
+                size_t pixelOffsetIn = pixelId * 4;
+                size_t pixelOffsetOut = pixelId * 3;
+                image.data[pixelOffsetOut] = image.data[pixelOffsetIn];
+                image.data[pixelOffsetOut + 1] = image.data[pixelOffsetIn + 1];
+                image.data[pixelOffsetOut + 2] = image.data[pixelOffsetIn + 2];
+            }
+            image.encoding = "rgb8";
```

Comment:
You defined `ImageEncodingNames`, I believe it makes sense to use it in case it changes one day (e.g. to capital letters).